api/base/serializers.py

```diff
@@ -25,6 +25,19 @@
 from website.project.model import has_anonymous_link
 
 
+def get_meta_type(serializer_class, request):
+    meta = getattr(serializer_class, 'Meta', None)
+    if meta is None:
+        return None
+    resource_type = getattr(meta, 'type_', None)
+    if resource_type is not None:
+        return resource_type
+    try:
+        return meta.get_type(request)
```

Comment:
What if you make `type_` a property where necessary, instead of using a method?

Reply:
The goal was to have type to depend on API version in cases where we need to change it, and thus use a method in place of the `type_` property in these cases.

We could have something like:
```
Meta:
    type_ = 'nodes'
    deprecated_type_ = 'linked_nodes'
```
and select the property based on API version in *this* method, but I was wanting to allow each serializer class to pin type changes to a specific version individually. A use case for that would be using what I've done here for #8792, which would be in a different API version.

Reply:
I right. I was thinking you could use a property method, but that wouldn't work without using our thread-local request:

```python
class Meta:
    @property
    def type_(self):
        request = api_globals.request
        if Version(request.version) > ...:
            ...
        else:
            ...
```

Probably not worth it, though. What you have here is fine.